src/huggingface_hub/hf_file_system.py

```diff
@@ -0,0 +1,431 @@
+                path_in_repo = path_in_repo_with_namespace
+                repo_and_revision_exist, err = self._repo_and_revision_exist(repo_type, repo_id, revision)
+                if not repo_and_revision_exist:
+                    if isinstance(err, (RepositoryNotFoundError, HFValidationError)):
+                        repo_id = repo_id_without_namespace
+                        path_in_repo = path_in_repo_without_namespace
+                        repo_and_revision_exist, _ = self._repo_and_revision_exist(repo_type, repo_id, revision)
+                        if not repo_and_revision_exist:
+                            raise FileNotFoundError(path) from err
+                    else:
+                        raise FileNotFoundError(path) from err
+        else:
+            repo_id = path
+            path_in_repo = ""
+            if "@" in path:
+                repo_id, revision_in_path = path.split("@", 1)
+                revision_in_path = unquote(revision_in_path)
+                revision = _align_revision_in_path_with_revision(revision_in_path, revision)
+            repo_and_revision_exist, _ = self._repo_and_revision_exist(repo_type, repo_id, revision)
+            if not repo_and_revision_exist:
+                raise NotImplementedError("Acces to repositories lists is not implemented.")
+
+        revision = revision if revision is not None else DEFAULT_REVISION
+        return ResolvedPath(repo_type, repo_id, revision, path_in_repo)
+
+    def invalidate_cache(self, path=None):
+        if not path:
+            self.dircache.clear()
+            self._repository_type_and_id_exists_cache.clear()
+        else:
+            path = self._strip_protocol(path)
+            path = self.resolve_path(path).unresolve()
+            while path:
+                self.dircache.pop(path, None)
+                path = self._parent(path)
+
+    def _open(
+        self,
+        path: str,
+        mode: str = "rb",
+        revision: Optional[str] = None,
+        **kwargs,
+    ):
+        if mode == "ab":
+            raise NotImplementedError("Appending to remote files is not yet supported.")
+        path = self._strip_protocol(path)
+        return HfFile(self, path, mode=mode, revision=revision, **kwargs)
+
+    def _rm(self, path, revision: Optional[str] = None, **kwargs):
+        path = self._strip_protocol(path)
+        resolved_path = self.resolve_path(path, revision=revision)
+        operations = [CommitOperationDelete(path_in_repo=resolved_path.path_in_repo)]
+        commit_message = f"Delete {path}"
+        self._api.create_commit(
+            repo_id=resolved_path.repo_id,
+            repo_type=resolved_path.repo_type,
+            token=self.token,
+            operations=operations,
+            revision=resolved_path.revision,
+            commit_message=kwargs.get("commit_message", commit_message),
+            commit_description=kwargs.get("commit_description"),
+        )
```

Comment:
Can be simplified a bit by using `HfApi.delete_file` here:

```suggestion
        self._api.delete_file(
            path_in_repo=resolved_path.path_in_repo,
            repo_id=resolved_path.repo_id,
            token=self.token,
            repo_type=resolved_path.repo_type,
            revision=resolved_path.revision,
            commit_message=kwargs.get("commit_message"),
            commit_description=kwargs.get("commit_description"),
        )
```